Update to latest `react-hook-form` and `@mantine/*` version

giammyisjammy opened the pull request:
Here's a PR that updates this package to use the latest updated versions of the packages in title.

I needed to make this update myself for another project I'm working on. So why not take the opportunity to release them for everyone?

More specifically, it address [this update of `@hookform/resolvers`](https://github.com/react-hook-form/resolvers/releases/tag/v5.0.0) which improves inference of input/output types from schema.

## What I did

- Run `npx npm-check-updates --interactive` to update `react-hook-form` and `@mantine/*` `devDependencies` version -> ended up bumping `vitest` and `zod` too due to blocking conflicts
- Bumped `@mantine/*` and `react-hook-form` versions in `peerDependencies`
- Re-created `package-lock.json` from scratch due to conflicting peer dependecis during `npm i` -> if anyone can suggest a better approach I'll gladly consider it
- Updated type definitions of components accordingly

I understand that a complete rewrite of the lockfile is a rather blunt solution. But again, if there are any concerns about it I'll eagerly discuss them and consider different suggestions.

giammyisjammy committed:
fix(types): update Switch component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update Textarea component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update TextInput component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update SwitchGroup component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update Autocomplete component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update Checkbox and CheckBoxGroup component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update Chip and ChipGroup component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update ColorInput component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update ColorPicker component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update DateInput component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update DatePickerInput component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update DateTimePicker component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update FileInput component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update Input component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update JsonInput component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update MonthPickerInput component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update MultiSelect component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update NativeSelect component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update NumberInput component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update PasswordInput component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update PinInput component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update Radio and RadioGroup component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update Rating component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update SegmentedControl component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update Select component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update Slider component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update TimeInput component type definitions for better TypeScript support

giammyisjammy committed:
fix(types): update YearPickerInput component type definitions for better TypeScript support

giammyisjammy committed:
chore(deps): bump @hookform/resolvers, @mantine/core, @mantine/dates, react-hook-form, vitest, and zod dependencies

giammyisjammy committed:
fix(types): fix Select onChange handler to match new type definition

johnmmackey commented:
@giammyisjammy this project looks dead - would you consider pushing your fork to NPM?

giammyisjammy commented:
> @giammyisjammy this project looks dead - would you consider pushing your fork to NPM?

I may. Never did this before, I'm in the process of learning the ropes.

In the meantime, if you absolutely want to try it, you may clone it and try it locally for yourself. Like so:

```bash
# Clone the project
❯ git clone https://github.com/giammyisjammy/react-hook-form-mantine.git
❯ cd react-hook-form-mantine

# Install deps and build it
❯ npm i && npm run build

# Create a tarball from package
❯ npm pack

# copy the tarball in your project directory
❯ cp react-hook-form-mantine-X.X.X.tgz ../your/project/location

# Install your dependency
❯ cd ../your/project/location && npm i ./react-hook-form-mantine-X.X.X.tgz
```
If you choose to do so, feedbacks are welcome of course.

giammyisjammy committed:
chore: update dependencies for Storybook and Vite

- upgraded @storybook/react-vite from ^7.4.2 to ^8.6.12
- upgraded @vitejs/plugin-react from ^4.0.4 to ^4.4.1
- upgraded storybook from ^7.4.2 to ^8.6.12
- upgraded vitest from ^3.1.1 to ^3.1.2
- updated vite to ^6.3.3

Resolves peer dependecies conflicts during npm install

giammyisjammy committed:
chore: update dependencies

- Updated @hookform/resolvers to version 5.0.1
- Updated @mantine/core and @mantine/dates to version 7.17.5
- Added dayjs as a new dependency with version 1.11.13
- Updated zod to version 3.24.3

giammyisjammy committed:
fix: minor fixes for tooling

giammyisjammy committed:
chore: format code

giammyisjammy committed:
chore: update example project

giammyisjammy committed:
chore: format code

giammyisjammy commented:
I'm running short of time to dedicate to this operation and there are still many things that I don't fully understand about the build process and dependencies management. Plus I still have some rough edges to work on i.e. I can't seem to make the type definitions work in the example project.

Bottom line: I'm not 100% confident about the quality of my fork. So I'd prefer to close this pull request for now.
